### PR TITLE
Feature/rover enum class

### PR DESCRIPTION
**Pull request by khancyr**
Add a bit of enum class on Rover. 

Override https://github.com/ArduPilot/ardupilot/pull/20011

gain some space

Add missing TERMINATE handler for fence breach and FS

### CODE REVIEW

File: Rover/fence.cpp [hunk 1]
```diff
@@ -18,32 +18,34 @@ void Rover::fence_check()
...
+                    set_mode(mode_hold, ModeReason::BATTERY_FAILSAFE);
                     break;
-                case Failsafe_Action_SmartRTL_Hold:
+                case FailsafeAction::SmartRTL_Hold:
                     if (!set_mode(mode_smartrtl, ModeReason::FENCE_BREACHED)) {
                         set_mode(mode_hold, ModeReason::FENCE_BREACHED);
                     }
                     break;
+                case FailsafeAction::Terminate:
+                    set_mode(mode_hold, ModeReason::FAILSAFE);
```
Comment (rmackay9) [inline Rover/fence.cpp:46]: I'm not sure we need to set the mode .. simply disarming is closer to what "terminate" means I think.

**Review by rmackay9 — APPROVED** [review]
Looks OK to me, thanks!

**Comment by rmackay9** [comment]
I've added to the Dev call but we could potentially merge before then.  @peterbarker or @IamPete1 may want to review as well although no big pressure.

File: Rover/fence.cpp (hunk 1 above)
Comment (khancyr) [reply Rover/fence.cpp:46]: Agree. I am not sure either.
My reasoning was the following : Terminate want the drone to stop and not move anymore. Disarm can work. 
But : Rover/boat have auto-arm (slope, river), CC can relaunch the arming and we will have endless issue, anything that will rearm will retrigger what cause the termination as the rover/boat will continue what it was doing when in non manual control. 
For those reasons, I add the hold so any rearm won't move the drone into a faulty/dangerous position, unless the operator explicitly ask for it.

Comment (rmackay9) [reply Rover/fence.cpp:46]: Re the possible interference with a companion computer, I don't think we should think too far ahead.  Let's just do the minimum and keep it simple.

Comment (khancyr) [reply Rover/fence.cpp:46]: ok

**Review by peterbarker — APPROVED** [review]

**Pull request merged**